pkg/s3repository/repository_protected_entity_type_manager.go

```diff
@@ -306,7 +308,11 @@ func (this *ProtectedEntityTypeManager) copyInt(ctx context.Context, sourcePEInf
 		peinfo: rPEInfo,
 	}
 
-	err = rpe.copy(ctx, dataReader, metadataReader)
+	_, err = rpe.DeleteSnapshot(ctx, id.GetSnapshotID())
```

Comment:
if there is no such a snapshot in s3PETM, it would always fail, right? E.g., for the very first copy of an upload, no previous snapshot in S3PETM.

Reply:
I got it. The S3.DeleteObject is used as a callback in ListObjectsPagesWithContext for each page listed. If there is no page, i.e. snapshot in our case, then no callback will be called. Good idea.

Reply:
Yeah, I thought it was cute.  No need to worry about dealing with the pagination by using this API